packages/common/src/createContract.ts

```diff
@@ -0,0 +1,125 @@
+import {
+  Abi,
+  Account,
+  Address,
+  Chain,
+  GetContractParameters,
+  GetContractReturnType,
+  Hex,
+  PublicClient,
+  SimulateContractParameters,
+  Transport,
+  WalletClient,
+  WriteContractParameters,
+  getContract,
+} from "viem";
+import pQueue from "p-queue";
+import pRetry from "p-retry";
+import { createNonceManager } from "./createNonceManager";
+import { debug as parentDebug } from "./debug";
+
+const debug = parentDebug.extend("createContract");
+
+// copied from viem because it isn't exported
+// TODO: import from viem?
```

Comment:
should we open a viem issue for this?

Reply:
<img width="704" alt="image" src="https://github.com/latticexyz/mud/assets/508855/23a0ae89-fff8-477b-b083-5a1eafe7d950">

🤷‍♂️